rosetta/docs/NATIVE_FP8.md

```diff
@@ -126,7 +124,7 @@ python -m paxml.main \
     ...
 ```
 
-Please ensure you include the first two flags, `--xla_gpu_enable_reduction_epilogue_fusion=false` and `--xla_gpu_enable_triton_gemm=false`, as they are essential for enabling the FP8 functionality. The additional flags primarily focus on performance enhancement and should also prove beneficial for non-FP8 executions.
+Please ensure you include the first flag, `--xla_gpu_enable_reduction_epilogue_fusion=false` as it is essential for enabling the FP8 functionality. The additional flags primarily focus on performance enhancement and should also prove beneficial for non-FP8 executions.
```

Comment:
Is this still true?
Having a flag absolutly needed for fp8 isn't great.
I'm pretty sure it isn't true anymore, or only in some cases.
@kaixih

Reply:
will wait for @kaixih 's comment

Reply:
No, this is out dated. I just confirmed we no longer need these flags. Actually, can we remove all these flags from the doc? ```
```
export XLA_FLAGS=" \
    --xla_gpu_enable_reduction_epilogue_fusion=false \
    --xla_gpu_enable_triton_gemm=false \
    --xla_gpu_enable_cudnn_fmha=false \
    --xla_gpu_enable_cudnn_layer_norm=true \
    --xla_gpu_enable_cublaslt=true \
    --xla_gpu_enable_latency_hiding_scheduler=true \
    --xla_gpu_enable_highest_priority_async_stream=true \
    --xla_gpu_all_reduce_combine_threshold_bytes=51200 "
```
I just tested the perf and using the default flags (==no flags) works fine.

Reply:
So, maybe merge this PR as is now and make a follow-up PR?
Big changes PRs can be difficult if there is a regression in corner case.

Reply:
Yes, agreed. I might be too optimistic about removing all the flags. It seems the `xla_gpu_enable_triton_gemm=false` is still necessary. I am working on a more complete test. Will file a PR after that.

Reply:
Why `xla_gpu_enable_triton_gemm` is still needed? Using it to fix hangs due to different autotuner results on different rank isn't needed anymore.

Reply:
Hi @kaixih I have been told that this flag has become a no-op... not sure how this affects the fp8 pattern matching! Let me keep this as is for now with `triton_gemm=false`. And we can make another PR once you have clarity on the flag status

Reply:
@kaixih can you double check that this flag don't do anything upstream and upstream is broken in some cases?

Reply:
@kocchop @nouiz 
Based on what we've found so far, can we update this section with the following set of flags:
```
export XLA_FLAGS=" \
    --xla_gpu_enable_triton_gemm=false \
    --xla_gpu_enable_pipelined_all_reduce=false \
    --xla_gpu_enable_pipelined_all_gather=false \
    --xla_gpu_enable_pipelined_reduce_scatter=false \
"
```
Here's the background: (1) The `xla_gpu_enable_triton_gemm` is back and the commit that changed it to a no-op has been reverted. (2) The Triton GEMM still needs to be explicitly turned off, otherwise it will break our fp8 pattern. (3) A recent change has made these `xla_gpu_enable_pipelined_xxx` enabled by default, which significantly slows down fp8 perf, even with (2) set.

(2) and (3) are tracked in our bug list and need to be addressed, but for now, both issues can be worked around.

Reply:
updated the changes @nouiz